qless/workers/forking.py

```diff
@@ -33,7 +38,7 @@ def stop(self, sig=signal.SIGINT):
                 logger.exception('Error stopping %s...' % cpid)
 
         # While we still have children running, wait for them
-        for cpid in self.sandboxes.keys():
+        for cpid in list(self.sandboxes):
```

Comment:
This loop modifies the dict while running (see the pop below) so you need to copy the keys. I changed the others for consistency.